### PR TITLE
AZP: Refactoring CI for use template

### PR DESCRIPTION
**Pull request by avildema**
need make template for use in other CI

### CODE REVIEW

File: buildlib/azure-pipelines-pr.yml [hunk 1]
```diff
@@ -19,396 +19,5 @@ pr:
...
-      - template: go/go-test.yml
-        parameters:
-          name: new
-          demands: ucx_new -equals yes
-      - template: go/go-test.yml
-        parameters:
-            name: gpu
-            demands: ucx_gpu -equals yes
+extends:
+  template: base_ci.yml
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:23]: pr/base_ci.yml

**Review by yosefe — APPROVED** [review]

File: buildlib/pr/main.yml [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+    - template: io_demo/io-demo.yml
+
+  - stage: jucx
+    dependsOn: [Static_check]
+    jobs:
+      - template: ../jucx/jucx-test.yml
+        parameters:
+          name: new
+          demands: ucx_new -equals yes
+      - template: ../jucx/jucx-test.yml
```
Comment (yosefe) [inline buildlib/pr/main.yml:378]: move jucx and go under pr/ ?

Comment (avildema) [reply buildlib/pr/main.yml:378]: we are using jucx in snapshot and release

Comment (yosefe) [reply buildlib/pr/main.yml:378]: ok

**Review by yosefe — APPROVED** [review]

**Pull request merged**